mongo/datadog_checks/mongo/api.py

```diff
@@ -114,10 +114,15 @@ def refresh_deployment_type(self):
             self.deployment_type = self._get_default_deployment_type()
         except Exception as e:
             self._log.debug(
-                "Unable to run `getCmdLineOpts`, got: %s. Assuming this is an Alibaba ApsaraDB instance.", str(e)
+                "Unable to run `getCmdLineOpts`, got: %s. Treating this is an Alibaba ApsaraDB instance.", str(e)
             )
-            # `getCmdLineOpts` is forbidden on Alibaba ApsaraDB
-            self.deployment_type = self._get_alibaba_deployment_type()
+            try:
+                self.deployment_type = self._get_alibaba_deployment_type()
+            except Exception as e:
+                self._log.debug(
+                    "Unable to detect Alibaba AsparaDB so switching to AWS DocumentDB, got error %s", str(e)
```

Comment:
can we add comments to highlight the exception is due to unsupported  admin command `shardingState` in AWS DocumentDB. we should probably also mention that `AWS DocumentDB Elastic Cluster` is not supported.

Reply:
AWS DocumentDB Elastic Cluster is not supported because elastic cluster only provides a cluster endpoint for connection, with no individual instance endpoints.

Reply:
@lu-zhengda if you don't mind I'll put the part about Elastic Cluster in the readme/docstrings. Not sure it makes sense in the error.

The point about shardingState command not being supported is well taken, I will adjust the message!

Reply:
Sounds good. readme is definitely the better to put what we support and what we do not support.